foundation-models/openai/src/test/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClientTest.java

```diff
@@ -128,122 +159,309 @@ void testErrorHandling() {
-                .getClassLoader()
-                .getResourceAsStream("chatCompletionResponse.json")
-                .readAllBytes());
-    stubFor(post(anyUrl()).willReturn(okJson(response)));
-
-    final var systemMessage =
-        new OpenAiChatMessage.OpenAiChatSystemMessage()
-            .setContent("You are a helpful, friendly and sometimes slightly snarky AI assistant!");
-    final var userMessage =
-        new OpenAiChatUserMessage().addText("Hello World! Why is this phrase so famous?");
-    final var request =
-        new OpenAiChatCompletionParameters().setMessages(List.of(systemMessage, userMessage));
-
-    final var result = client.chatCompletion(request);
-
-    assertThat(result).isNotNull();
-    assertThat(result.getCreated()).isEqualTo(1719300073);
-    assertThat(result.getId()).isEqualTo("chatcmpl-9dumHtDEyysGFnknk17n4Lt37tg7T");
-    assertThat(result.getModel()).isEqualTo("gpt-4-32k");
-    assertThat(result.getObject()).isEqualTo("chat.completion");
-    assertThat(result.getSystemFingerprint()).isNull();
-
-    assertThat(result.getUsage().getCompletionTokens()).isEqualTo(54);
-    assertThat(result.getUsage().getPromptTokens()).isEqualTo(13);
-    assertThat(result.getUsage().getTotalTokens()).isEqualTo(67);
-
-    assertThat(result.getPromptFilterResults()).hasSize(1);
-    OpenAiPromptFilterResult promptFilterResults = result.getPromptFilterResults().get(0);
-    assertThat(promptFilterResults.getPromptIndex()).isEqualTo(0);
-    assertThat(promptFilterResults.getContentFilterResults().getSexual().isFiltered()).isFalse();
-    assertThat(promptFilterResults.getContentFilterResults().getSexual().getSeverity())
-        .isEqualTo(SAFE);
-    assertThat(promptFilterResults.getContentFilterResults().getViolence().isFiltered()).isFalse();
-    assertThat(promptFilterResults.getContentFilterResults().getViolence().getSeverity())
-        .isEqualTo(SAFE);
-    assertThat(promptFilterResults.getContentFilterResults().getHate().isFiltered()).isFalse();
-    assertThat(promptFilterResults.getContentFilterResults().getHate().getSeverity())
-        .isEqualTo(SAFE);
-    assertThat(promptFilterResults.getContentFilterResults().getSelfHarm()).isNull();
-    assertThat(promptFilterResults.getContentFilterResults().getProfanity()).isNull();
-    assertThat(promptFilterResults.getContentFilterResults().getError()).isNull();
-    assertThat(promptFilterResults.getContentFilterResults().getJailbreak().isFiltered()).isFalse();
-    assertThat(promptFilterResults.getContentFilterResults().getJailbreak().isDetected()).isFalse();
-
-    assertThat(result.getChoices()).hasSize(1);
-    OpenAiChatCompletionChoice choice = result.getChoices().get(0);
-    assertThat(choice.getFinishReason()).isEqualTo("stop");
-    assertThat(choice.getIndex()).isEqualTo(0);
-    assertThat(choice.getMessage().getContent())
-        .isEqualTo(
-            """
+  void chatCompletion() throws IOException {
+    try (var inputStream =
```

Comment:
why change this? with the additional indentation from the try it looks a bit less pretty than before arguably

Reply:
It removes a warning from my IDE. I read all warnings before committing.

Reply:
maybe you can disable the warning for tests? Because here on error the test fails anyway, so closing the stream isn't really necessary..